src/net.cpp

```diff
@@ -1475,6 +1475,9 @@ void CConnman::ThreadDNSAddressSeed()
                     vAdd.push_back(addr);
                     found++;
                 }
+                vAdd.erase(std::remove_if(vAdd.begin(), vAdd.end(),
+                                          [](const CAddress& addr) { return !IsReachable(addr); }),
```

Comment:
e91e3efac9c7b845fbabe30f23a1981706194ccb: why not just check this condition before calling `vAdd.push_back(addr)`? We should probably not increment `found` either when we don't use an address.

Reply:
Good point. On second thought, the filtering seems unnecessary in the DNS seed case, because with the second commit, we will skip querying the DNS seeds if IPv4 and IPv6 are unreachable, while DNS seeds only return IPv4 and IPv6 addresses. So I removed it.